ssg/build_cpe.py

```diff
@@ -189,72 +206,74 @@ def to_xml_element(self):
 
         return cpe_test
 
-    def enrich_with_cpe_info(self, cpe_products):
+    def add_enriched_cpe_items(self, product_cpes):
         for arg in self.args:
-            arg.enrich_with_cpe_info(cpe_products)
+            arg.add_enriched_cpe_items(product_cpes)
 
-    def to_bash_conditional(self):
-        cond = ""
+    def get_remediation_conditional(self, language, product_cpes, conditionals_path, env_yaml):
+        contents = ''
+        config = defaultdict(lambda: None)
+        op = " "
         if self.is_not():
-            cond += "! "
-            op = " "
-        cond += "( "
-        child_bash_conds = [
-            a.to_bash_conditional() for a in self.args
-            if a.to_bash_conditional() != '']
+            contents += CONDITIONAL_OPERATORS[language]["not"] + " "
+        contents += "( "
+        child_condlines = []
+        for a in self.args:
+            r = a.get_remediation_conditional(language, product_cpes, conditionals_path, env_yaml)
+            if r is not None:
+                cont = r.contents.strip()
+                if cont:
+                    child_condlines.append(cont)
         if self.is_or():
-            op = " || "
+            op = " " + CONDITIONAL_OPERATORS[language]["or"] + " "
         elif self.is_and():
-            op = " && "
-        cond += op.join(child_bash_conds)
-        cond += " )"
-        return cond
-
-    def to_ansible_conditional(self):
-        cond = ""
-        if self.is_not():
-            cond += "not "
-            op = " "
-        cond += "( "
-        child_ansible_conds = [
-            a.to_ansible_conditional() for a in self.args
-            if a.to_ansible_conditional() != '']
-        if self.is_or():
-            op = " or "
-        elif self.is_and():
-            op = " and "
-        cond += op.join(child_ansible_conds)
-        cond += " )"
-        return cond
+            op = " " + CONDITIONAL_OPERATORS[language]["and"] + " "
+        contents += op.join(child_condlines)
+        contents += " )"
+        return namedtuple('remediation', ['contents', 'config'])(contents=contents, config=config)
 
-
-class CPEALFactRef (Symbol):
+    
+class CPEALFactRef(Symbol):
 
     prefix = "cpe-lang"
     ns = PREFIX_TO_NS[prefix]
 
     def __init__(self, obj):
         super(CPEALFactRef, self).__init__(obj)
         self.cpe_name = obj  # we do not want to modify original name used for platforms
-        self.bash_conditional = ""
-        self.ansible_conditional = ""
-
-    def enrich_with_cpe_info(self, cpe_products):
-        self.bash_conditional = cpe_products.get_cpe(self.cpe_name).bash_conditional
-        self.ansible_conditional = cpe_products.get_cpe(self.cpe_name).ansible_conditional
-        self.cpe_name = cpe_products.get_cpe_name(self.cpe_name)
+        self.conditional = {}
+
+    def add_enriched_cpe_items(self, product_cpes):
```

Comment:
OK. So currently it works like this when creating templated platforms:
The build system goes through all rule.yml files and collects all platforms. they might be composed of more logically connected platform definitions. These definitions are then matched agains files in shared/applicability. You can notice that some files such as "package.yml" have placeholders inside. They are there usually to create correct CPE IDs and CPE names. At this moment the enriching happens - these placeholders are filled in with parameters collected from rule.yml (package name, version...).
This happens only once - when the rule.ymls are read for the first time. then the CPE items are saved within build/$product/cpe_items and this functionality is no longer used.
I admit that the implementation might not be very clear but we did not find a better way. The purpose is to create individual CPE items with individual IDs as soon as possible so that they can be later processed by the build system.
Is it clearer now?